File: app/src/main/java/me/devsaki/hentoid/parsers/images/HdPornComicsParser.java
package me.devsaki.hentoid.parsers.images;

import static me.devsaki.hentoid.util.network.HttpHelper.getOnlineDocument;

import androidx.annotation.NonNull;
import androidx.core.util.Pair;

import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

import java.util.ArrayList;
import java.util.List;

import me.devsaki.hentoid.database.domains.Content;
import me.devsaki.hentoid.parsers.ParseHelper;
import me.devsaki.hentoid.util.exception.ParseException;

public class HdPornComicsParser extends BaseImageListParser {

    @Override
    protected List<String> parseImages(@NonNull Content content) throws Exception {
        // Fetch the book gallery page
        Document doc = getOnlineDocument(content.getGalleryUrl());
        if (null == doc)
            throw new ParseException("Document unreachable : " + content.getGalleryUrl());

        List<Element> pages = doc.select("figure a picture img");

        return parseImages(pages);
    }

    public static List<String> parseImages(@NonNull List<Element> pages) {
        List<String> result = new ArrayList<>();

        for (Element e : pages) {
            String href = ParseHelper.getImgSrc(e);
            result.add(href.replace("thumbs", "uploads"));
        }

        return result;
    }

    @Override
    protected List<String> parseImages(@NonNull String chapterUrl, String downloadParams, List<Pair<String, String>> headers) throws Exception {
        // Nothing because no chapters for this source
        return null;
    }

}

File: app/src/main/java/me/devsaki/hentoid/parsers/images/HdPornComicsParser.kt
package me.devsaki.hentoid.parsers.images

import me.devsaki.hentoid.database.domains.Content
import me.devsaki.hentoid.parsers.getImgSrc
import me.devsaki.hentoid.util.exception.ParseException
import me.devsaki.hentoid.util.network.getOnlineDocument
import org.jsoup.nodes.Element

class HdPornComicsParser : BaseImageListParser() {
    companion object {
        fun parseImages(pages: List<Element>): List<String> {
            return pages.map { getImgSrc(it) }.map { it.replace("thumbs", "uploads") }
        }
    }

    override fun parseImages(content: Content): List<String> {
        // Fetch the book gallery page
        val doc = getOnlineDocument(content.galleryUrl)
            ?: throw ParseException("Document unreachable : " + content.galleryUrl)

        return parseImages(doc.select("figure a picture img").filterNotNull())
    }
}